src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -914,5 +983,32 @@ private static string CreateDeterminismKey(CommandLineArguments args, string[] r
 
             return builder.ToString();
         }
+
+
+        /// <summary>
+        /// The set of source file paths that are in the set of embedded paths.
+        /// This is used to prevent reading source files that are embedded twice.
+        ///
+        /// It further prevents a race where the file we embed could have changed
+        /// since the time it was parsed.
+        /// </summary>
+        protected IReadOnlySet<string> EmbeddedSourcePaths { get; }
```

Comment:
Where is this used (can't find it). Careful when using sets - make sure the order doesn't influence the results of a compilation.

---
In reply to: [72847645](https://github.com/dotnet/roslyn/pull/12353#discussion_r72847645) [](ancestors = 72847645)